src/app/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useRef, useEffect } from "react";
+import { clickOutsideHandler } from "./hooks";
+import { Icon } from "app/components/Icon";
+
+export type Option = {
+	label: string;
+	value: string | number;
+};
+
+type Props = {
+	as?: React.ElementType;
+	children?: React.ReactNode;
+	onSelect?: any;
+	options?: Option[];
+	toggleIcon: string;
+};
+
+/*
+ * Dropdown options list
+ */
+const renderOptions = (options: Option[], onSelect: any) => (
+	<ul data-testid="dropdown__options">
+		{options.map((option: Option, key: number) => (
+			<li
+				className="block cursor-pointer px-8 py-4 text-sm font-semibold text-theme-neutral-800 hover:bg-theme-neutral-200 hover:text-theme-primary-600"
+				key={key}
+				data-testid={`dropdown__option--${key}`}
+				onClick={() => onSelect(option)}
+			>
+				{option.label}
+			</li>
+		))}
+	</ul>
+);
+
+export const Dropdown = ({ children, options, onSelect, toggleIcon }: Props) => {
+	const [isOpen, setIsOpen] = useState(false);
+
+	/*
+	 * Toggle & hide dropdown content.
+	 */
+	const toggle = () => setIsOpen(!isOpen);
+	const hide = () => setIsOpen(false);
+
+	/*
+	 * Hide dropdown upon option click.
+	 * Call `onSelect` and close dropdown.
+	 */
+	const select = (option: Option) => {
+		setIsOpen(false);
+		if (typeof onSelect === "function") onSelect(option);
+	};
+
+	/*
+	 * Hook to hide dropdown on outside click event.
+	 */
+	const ref = useRef(null);
+	useEffect(() => clickOutsideHandler(ref, hide), [ref]);
```

Comment:
Same as `toggle` and `hide`. If you feel the need for a comment because the method names or code give no indication of what is happening you should rename them or make it more clear otherwise.

Reply:
Not necessary indeed. Removed them.